lambda/switch.ml

```diff
@@ -266,25 +266,19 @@ let pret chan = function
           for i = 0 to len1-2 do
             r.(i) <- c1.(i)
           done ;
-
           let l =
-            if len1-2 >= 0 then begin
```

Comment:
Since there's already a match earlier that got rid of the case `len1 = 0`, this new test is equivalent to `len1 = 1`. I don't mind if you want to keep the test as it is, but I would find it much more helpful if you added this as a comment on the `then` branch than the comment you've put on the else branch. (The same applies below too).

Reply:
This particular part of the code does not rely on the fact that `len1, len2 > 0`, but it does rely on the len>=2 conditions for correctness of array indexing. I'm not sure why it would be helpful to recall the `> 0` condition here for that reason. The comment is there to relate the condition to the correctness of the array indexing right below. Luc rather asked to keep the tests in their unintuitive previous form, and I was trying to find a compromise with the comment... you are not making my life simpler :D